FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90

```diff
@@ -250,6 +266,123 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
       ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%uswsfci(:)
     enddo
 
+    if (Model%override_surface_radiative_fluxes) then
+      idx = idx + 1
+      ExtDiag(idx)%axes = 2
+      ExtDiag(idx)%name = 'DLWRF_from_rrtmg'
+      ExtDiag(idx)%desc = 'surface downward longwave flux due to RRTMG'
+      ExtDiag(idx)%unit = 'W/m**2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%cnvfac = cn_one
+      ExtDiag(idx)%time_avg = .TRUE.
+      ExtDiag(idx)%intpl_method = 'bilinear'
+      ExtDiag(idx)%coarse_graining_method = 'area_weighted'
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%dlwsfc_rrtmg(:)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 2
+      ExtDiag(idx)%name = 'DLWRFI_from_rrtmg'
+      ExtDiag(idx)%desc = 'instantaneous surface downward longwave flux due to RRTMG'
+      ExtDiag(idx)%unit = 'W/m**2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%intpl_method = 'bilinear'
+      ExtDiag(idx)%coarse_graining_method = 'area_weighted'
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%dlwsfci_rrtmg(:)
+      enddo
+
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 2
+      ExtDiag(idx)%name = 'ULWRF_from_rrtmg'
+      ExtDiag(idx)%desc = 'surface upward longwave flux due to RRTMG'
+      ExtDiag(idx)%unit = 'W/m**2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%cnvfac = cn_one
+      ExtDiag(idx)%time_avg = .TRUE.
+      ExtDiag(idx)%intpl_method = 'bilinear'
+      ExtDiag(idx)%coarse_graining_method = 'area_weighted'
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%ulwsfc_rrtmg(:)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 2
+      ExtDiag(idx)%name = 'ULWRFI_from_rrtmg'
+      ExtDiag(idx)%desc = 'instantaneous surface upward longwave flux due to RRTMG'
+      ExtDiag(idx)%unit = 'W/m**2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%intpl_method = 'bilinear'
+      ExtDiag(idx)%coarse_graining_method = 'area_weighted'
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%ulwsfci_rrtmg(:)
+      enddo
```

Comment:
Since we do not override the upward longwave radiative flux at the surface, technically these diagnostics will always be the same as the existing fortran ones (i.e. they are basically duplicate diagnostics; `ULWRF` = `ULWRF_from_rrtmg` and `ULWRFI` = `ULWRFI_from_rrtmg`).  I can see an argument for or against adding them, so I left them in for now.  I'd be happy to remove them however.

Reply:
I think it's fine to have duplicated diagnostics.